libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/VoiceInstructionMilestone.java

```diff
@@ -15,21 +16,28 @@
   private DirectionsRoute currentRoute;
   private LegStep currentStep;
   private List<VoiceInstructions> stepVoiceInstructions;
+  private final VoiceInstructionLoader voiceInstructionLoader;
 
   VoiceInstructionMilestone(Builder builder) {
     super(builder);
+    voiceInstructionLoader = VoiceInstructionLoader.getInstance();
   }
 
   @Override
   public boolean isOccurring(RouteProgress previousRouteProgress, RouteProgress routeProgress) {
```

Comment:
IMO `isOccurring` is getting too long.
What about [decomposing these conditionals](https://sourcemaking.com/refactoring/decompose-conditional) extracting blocks of code into private methods and make intention clearer?

We should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).

Reply:
I think this is out of scope, I just barely touched that method in this PR